cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/transforms/StreamToStructuredRecordTransform.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.templates.etl.transforms;
+
+import co.cask.cdap.api.data.format.FormatSpecification;
+import co.cask.cdap.api.data.format.RecordFormat;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.flow.flowlet.StreamEvent;
+import co.cask.cdap.templates.etl.api.Emitter;
+import co.cask.cdap.templates.etl.api.Property;
+import co.cask.cdap.templates.etl.api.StageConfigurer;
+import co.cask.cdap.templates.etl.api.Transform;
+import co.cask.cdap.templates.etl.transforms.formats.RecordFormats;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import org.apache.hadoop.io.LongWritable;
+
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * Transforms {@link StreamEvent} to {@link StructuredRecord}
+ */
+public class StreamToStructuredRecordTransform extends Transform<LongWritable, StreamEvent,
+                                                                 LongWritable, StructuredRecord> {
+  private static final String SCHEMA = "schema";
+  private static final String SCHEMA_TYPE = "schemaType";
+  private static SchemaWrapper schemaWrapper = null;
+
+  @Override
+  public void configure(StageConfigurer configurer) {
+    configurer.setName(StreamToStructuredRecordTransform.class.getName());
+    configurer.setDescription("Transforms a StreamEvent to StructuredRecord.");
+    configurer.addProperty(new Property(SCHEMA, "The schema of the body of stream events", true));
+    configurer.addProperty(new Property(SCHEMA_TYPE, "Type of the Schema ex: CSV, TSV etc.", true));
+  }
+
+  @Override
+  public void transform(@Nullable LongWritable inputKey, StreamEvent streamEvent,
+                        Emitter<LongWritable, StructuredRecord> emitter) throws Exception {
+
+    if (schemaWrapper == null) {
```

Comment:
hm if runtime args are not available in initialize that seems like a bug in the framework